feat: Added way to get GlobalObjectIdHash from a NetworkPrefab [MTT-1645]

TwoTenPvP opened the pull request:
<!-- Replace this block with what this PR does and why. Describe what you'd like reviewers to know, how you applied the engineering principles, and any interesting tradeoffs made. Delete bullet points below that don't apply, and update the changelog section as appropriate. -->

<!-- Add short version of the JIRA ticket to the PR title (e.g. "feat: new shiny feature [MTT-123]") -->
MTT-1645

<!-- Add RFC link here if applicable. -->

## Changelog

- Added a way to get the `GlobalObjectIdHash` from a GameObject for use in `ConnectionApproval`.

## Testing and Documentation

- Includes unit tests.

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

TwoTenPvP committed:
feat: Added way to get GlobalObjectIdHash from a NetworkPrefab

TwoTenPvP committed:
Added changelog entry

TwoTenPvP committed:
Merge branch 'develop' into get-global-object-id-hash

NoelStephensUnity committed:
Merge branch 'develop' into get-global-object-id-hash

NoelStephensUnity reviewed (approved):
I think GetGlobalObjectIdHash might be better named something else to signify that it only allows registered NetworkPrefabs.
Other than that minor name thing... everything else LGTM!

TwoTenPvP committed:
Changed signature

TwoTenPvP committed:
Merge branch 'get-global-object-id-hash' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into get-global-object-id-hash

NoelStephensUnity committed:
fix

This fixes the manual and integration tests that use connection approval and a player prefab override.

JesseOlmer commented:
Closed and re-opened to fix the JIRA association. Please make sure to add the JIRA ticket to the title upon PR creation in the future.
